six/modules/c++/cphd/include/cphd/Wideband.h

```diff
@@ -140,6 +157,41 @@ class Wideband
              lastSample, numThreads, buffer);
     }
 
+    /*
+     * Read all channel, vector(s), and sample(s)
+     * Performs endian swapping if necessary
+     *
+     * \param channel 0-based channel
+     * \param firstVector 0-based first vector to read (inclusive)
+     * \param lastVector 0-based last vector to read (inclusive).  Use ALL to
+     * read all vectors
+     * \param firstSample 0-based first sample to read (inclusive)
+     * \param lastSample 0-based last sample to read (inclusive).  Use ALL to
+     * read all samples
+     * \param numThreads Number of threads to use for endian swapping if
+     * necessary
+     * \param data Will contain the read in data.  Throws if buffer has not
+     * been allocated to a sufficient size
+     * (numChannels * numVectors * numSamples * elementSize)
+     */
+    void readAll(size_t firstVector,
```

Comment:
Not sure about the correct usecase of wideband read.
Would it make sense to read all (numChannel) signal arrays with the same start vector/sample and end vector/sample?
Or should my input parameters be vectors of size num channels for each of those parameter?

Reply:
From my (admittedly very limited) understanding of channels here, concatenating all the channels together isn't necessarily a useful thing to want to do. Let's just make them read each channel individually for now (I think most stuff is single-channel anyway?), and if it turns out everyone wants to handle multi-channel reading in the same way, it's easy enough to add it when they ask for it.

Reply:
Sounds good!